statsmodels/tsa/holtwinters.py

```diff
@@ -0,0 +1,444 @@
+    if alpha*beta==0.0: return max_seen
+    if beta > alpha or gamma > 1-alpha: return max_seen
+    for i in range(1, n):
+        l[i] = (y_alpha[i-1]) - (alpha * s[i-1]) + (alphac * (l[i-1] * b[i-1]**phi))
+        b[i] = (beta * (l[i] / l[i-1])) + (betac * b[i-1]**phi)
+        s[i+m-1] = y_gamma[i-1] - (gamma * (l[i-1] * b[i-1]**phi)) + (gammac * s[i-1])
+    return sqeuclidean((l*phi*b)+s[:-(m-1)], y)        
+
+def holt_winters(data, alpha=None, beta=None, gamma=None, m=None, h=1, trend=None, 
+                 damped=False, seasonal=None, phi=None, optimized=True,
+                 use_boxcox=False, remove_bias=False, use_basinhopping=False):  
+    """
+    Holt Winter's Exponential Smoothing
+
+    Parameters
+    ----------
+    data : array-like
+        Time series
+    alpha : float, optional
+        The alpha value of the simple exponential smoothing, if the value is
+        set then this value will be used as the value.
+    beta :  float, optional
+        The beta value of the holts trend method, if the value is
+        set then this value will be used as the value.
+    gamma : float, optional
+        The gamma value of the holt winters seasonal method, if the value is
+        set then this value will be used as the value.
+    m : int, optional
+        The number of seasons to consider for the holt winters.
+    h : int, optional
+        The number of time steps to forecast ahead.
+    trend : {"add", "mul", None}, optional
+        Type of trend component.
+    damped : bool, optional
+        Should the trend component be damped.
+    seasonal : {"add", "mul", None}, optional
+        Type of seasonal component.
+    phi : float, optional
+        The phi value of the damped method, if the value is
+        set then this value will be used as the value.
+    optimized : bool, optional
+        Should the values that have not beem set above be optimized automatically?
+    use_boxcox : {True, False, 'log'}, optional
+        Should the boxcox tranform be applied to the data first? If 'log' then 
+        apply the log.
+    remove_bias : bool, optional
+        Should the bias be removed from the fcast and fitted values before being
+        returned?
+    use_basinhopping : bool, optional
+        Should the opptimser try harder using basinhopping to find optimal values? 
+
+    Returns
+    -------
+    results : namedtuple
+        A namedtuple with fcast, fitted and other attributes
+
+    Notes
+    -----
+    This is a full implementation of the holt winters exponential smoothing as 
+    per [1]. This includes all the unstable methods as well as the stable methods.
+    The implementaion of the library follows as per the R library as much as possible
+    whilest still being pythonic.
+    
+    References
+    ----------
+    [1] Hyndman, Rob J., and George Athanasopoulos. Forecasting: principles and practice. OTexts, 2014.
+    """
+    opt=None
```

Comment:
Missing a space.   Might as well autopep8 this file since new.

Reply:
done